flatfs.go

```diff
@@ -1042,11 +1071,30 @@ func (fs *Datastore) walk(path string, result *query.ResultBuilder) error {
 			continue
 		}
 
+		var (
+			value []byte
+			size  int
+			err   error
+		)
+		if !result.Query.KeysOnly {
+			value, err = ioutil.ReadFile(filepath.Join(path, fn))
+			size = len(value)
```

Comment:
Is it safe to assume that if there's an error, value will have length 0? (seems like it should be, just want to confirm)

Reply:
Hm. Good question. No. It'll return the partial value and an error.  I'll change this.